operator/pkg/legacy/secrets.go

```diff
@@ -105,29 +105,38 @@ func getUseExisting(vals chartutil.Values, key string) ([]string, error) {
 
 func (i *injector) enrich(ctx context.Context, vals chartutil.Values, key string, secretNames []string, namespaceName string) (chartutil.Values, error) {
```

Comment:
Unrelated to your PR, but do you know what is `secretNames` here ? e.g. What are we doing in this section?

Reply:
secretNames in this context are names of secrets that are known to be used to set pull secrets. AFAIK this is mainly used for dev work/CI/upstream build where you have to specify pull secrets manually.

It looks for common names used in tooling e.g a secret called `stackrox` or `mainImagePullSecrets` and sets appropriate helm values to use those secrets as pull secrets for the deployments.